GraphBase/src/graph/Vertex.java

```diff
@@ -0,0 +1,49 @@
+package graph;
+
+import java.util.Map;
+import java.util.HashMap;
+import java.util.NoSuchElementException;
+
+public class Vertex {
+
+  private int vertexId;
+  private Map<Integer, Edge> adjEdges;
+
+  public Vertex(int newVertexId) {
+    this.vertexId = newVertexId;
+    this.adjEdges = new HashMap<Integer, Edge>();
+  }
+
+  public int getId() {
+    return vertexId;
+  }
+
+  public Map<Integer, Edge> getAdjEdges() {
+    return adjEdges;
+  }
+
+  public boolean addFromAdjEdge(Edge newEdge) {
+    if (newEdge.getFromId() != vertexId) {
+      return false; // (represents the edge is not from the vertex)
+    }
+    adjEdges.put(newEdge.getToId(), newEdge);// use the tovertex'id to be the newedge's id;
+    return true;// normal case
+  }
+
+  public boolean addToAdjEdge(Edge newEdge) {
+    if (newEdge.getToId() != vertexId) {
+      return false;//(represents the edge is not to this vertex)
+    }
+    adjEdges.put(newEdge.getFromId(), newEdge);
+    return true;
+  }
+
+  public Edge deleteAdjEdge(Integer edgeId) {
```

Comment:
Your edge seems to be either indexed via source id or to id, why there is edgeId?